modules/basics/src/main/java/com/opengamma/basics/date/ImmutableHolidayCalendar.java

```diff
@@ -141,11 +156,33 @@ private ImmutableHolidayCalendar(String name, SortedSet<LocalDate> holidays, Set
     this.holidays = ImmutableSortedSet.copyOfSorted(holidays);
     this.weekendDays = Sets.immutableEnumSet(weekendDays);
     if (holidays.isEmpty()) {
-      range = LocalDateRange.ALL;
+      this.range = LocalDateRange.ALL;
+      this.startYear = 0;
+      this.lookup = new int[0];
     } else {
       this.range = LocalDateRange.ofClosed(
           holidays.first().with(TemporalAdjusters.firstDayOfYear()),
           holidays.last().with(TemporalAdjusters.lastDayOfYear()));
+      this.startYear = range.getStart().getYear();
+      int[] array = new int[(range.getEndExclusive().getYear() - startYear) * 12];
+      for (LocalDate date : holidays) {
+        int index = (date.getYear() - startYear) * 12 + date.getMonthValue() - 1;
```

Comment:
May become clearer later, but given you have enough space (just!) is it worth aligning the dayOfMonth with the bit position, ignoring the 0 bit?